Modules/_asynciomodule.c

```diff
@@ -1461,16 +1462,36 @@ _asyncio_Task___init___impl(TaskObj *self, PyObject *coro, PyObject *loop)
 /*[clinic end generated code: output=9f24774c2287fc2f input=8d132974b049593e]*/
 {
     PyObject *res;
+    int tmp;
     _Py_IDENTIFIER(add);
 
     if (future_init((FutureObj*)self, loop)) {
         return -1;
     }
 
+    if (!PyCoro_CheckExact(coro)) {
+        // fastpath failed, perfom slow check
+        // raise after Future.__init__(), attrs are required for __del__
+        res = PyObject_CallFunctionObjArgs(asyncio_iscoroutine_func,
+                                           coro, NULL);
+        if (res == NULL) {
+            return -1;
+        }
+        tmp = PyObject_Not(res);
+        Py_DECREF(res);
+        if (tmp < 0) {
+            return -1;
+        }
+        if (tmp) {
+            self->task_log_destroy_pending = 0;
+            PyErr_Format(PyExc_TypeError, "%R is not a coroutine", coro, NULL);
```

Comment:
Can we change the error message to "a coroutine was expected, got %R" in both C/Python implementations?